src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,99 @@
+package duke.storage;
+
+import duke.DukeException;
+import duke.parser.Parser;
+import duke.task.*;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.Path;
+import java.util.ArrayList;
+
+public class Storage {
+    public static final String DEFAULT_STORAGE_DIRECTORY = "./data/";
+    public static final String DEFAULT_STORAGE_FILEPATH = DEFAULT_STORAGE_DIRECTORY + "tasks.txt";
+
+    public final Path path;
+
+    public Storage() throws DukeException {
+        this(DEFAULT_STORAGE_FILEPATH);
+    }
+
+    public Storage(String fileName) throws DukeException {
+        path = Paths.get(DEFAULT_STORAGE_DIRECTORY + fileName);
+        if (isValidPath(path)) {
+            throw new DukeException("duke.storage.Storage file should end with '.txt'");
+        }
+    }
+
+    public ArrayList<Task> loadALlTasks() throws DukeException  {
```

Comment:
Minor mistake found in function name.

```suggestion
    public ArrayList<Task> loadAllTasks() throws DukeException  {
```